Review followup: PR #20612 — t2766: wire greeting-cache-helper into Claude Code runtime

## Unaddressed review bot suggestions

PR #20612 was merged with unaddressed review bot feedback. Each comment
below includes its file path, line number, a direct link to the inline
review comment, and a `diff` fence with the code context the bot was
flagging. Resolved and outdated threads are filtered out via GitHub's
GraphQL review-thread state. Read the relevant lines, decide whether
the suggestion is correct, and either apply the fix or close this issue
with a wontfix rationale.

**Source PR:** https://github.com/marcusquinn/aidevops/pull/20612

---

### You are the triager (worker-is-triager rule)

This issue is **auto-created from review bot output** and dispatched
directly to you. Review bots can be wrong: hallucinated line refs, false
premises about codebase structure, template-driven sweeps without
measurements (see GH#17832-17835 for prior art and `prompts/build.txt`
section 6a). **Do not assume the bot is correct.** Verify before acting.

You must end in exactly one of three outcomes — no fourth "hand it back
to the human" path exists. Humans approve decisions; they do not re-do
analysis.

#### Outcome A — Premise falsified → close the issue

1. Read the cited `file:line` (listed under *Files to modify* below).
2. If the bot's claim is factually wrong (file doesn't exist at that
   line, function doesn't behave as described, "auto-generated" section
   isn't actually auto-generated, etc.), **close the issue** with a
   comment in this shape:

   > **Premise falsified.** \<what the bot claimed\>. \<what the code
   > actually shows, with a `file:line` citation or one-line quote\>.
   > Not acting.

   No PR. No further dispatch. The closing comment trains the next
   session reading this thread and the noise filter.

#### Outcome B — Premise correct + fix is obvious → implement and PR

1. Verify the bot's premise as above.
2. Read the Worker Guidance section below, open a worktree, implement.
3. Open a PR with `Resolves #<this-issue-number>` in the body
   (use THIS issue's number, not the source PR's) so merge auto-closes it.
4. Follow the normal Lifecycle Gate (brief, tests, review-bot-gate,
   merge, postflight).

#### Outcome C — Premise correct but approach is a genuine judgment call

Only use this path if you reach it after Outcomes A and B don't apply:
the bot's finding is real, but the fix requires a decision that is
architectural, policy, breaking-change, or otherwise genuinely outside
what you can resolve autonomously. In that case, post a **decision
comment** with exactly these fields:

- **Premise check:** one line, confirming the finding is real.
- **Analysis:** 2-4 bullets on the trade-offs.
- **Recommended path:** the option you would take if the decision were
  yours, with rationale.
- **Specific question:** the single decision the human needs to make
  (yes/no or pick-one, not open-ended).

Then apply `needs-maintainer-review` and stop. The human wakes up to a
ready-to-approve recommendation, not a blank task.

> **Ambiguity about scope or style is not Outcome C.** Per
> `prompts/build.txt` "Reasoning responsibility", the model does the
> thinking and delivers a recommendation. Only escalate what is genuinely
> a maintainer-only decision.

### Worker Guidance

**Files to modify:**

- `.agents/scripts/generate-runtime-config.sh:152`
- `.agents/scripts/generate-runtime-config.sh:159`

**Implementation steps (Outcome B path):**

1. Read the `diff` block under each inline comment below — it shows the
   exact code the bot was flagging. Open the file only if you need
   surrounding context beyond what the diff tail shows.
2. Read the bot's full comment below the diff — it contains the rationale
   and any suggested change.
3. Verify the premise before implementing (see Outcome A). If the premise
   is wrong, switch to Outcome A instead of burning iterations trying to
   satisfy a wrong suggestion.
4. If multiple comments target the same file, group your edits into one
   logical commit.
5. Run `shellcheck` / `markdownlint-cli2` / project tests as appropriate.

**Verification:**

- Open the new PR with `Resolves #<this-issue>` so this followup is auto-closed on merge.
- If the bot's suggestion was incorrect, close this issue with a Outcome A comment — do not open a no-op PR.

### Inline comments

#### gemini-code-assist on `.agents/scripts/generate-runtime-config.sh:152`

[View inline comment](https://github.com/marcusquinn/aidevops/pull/20612#discussion_r3132657180)

```diff
+		# shellcheck disable=SC2088
+		cache_path="~/.aidevops/cache/session-greeting.txt"
+		plugin_name="opencode-aidevops plugin"
+		# shellcheck disable=SC2088
+		global_config_path="~/.config/opencode/opencode.json"
+		;;
+	claude-code)
+		# shellcheck disable=SC2088
+		cache_path="~/.aidevops/cache/session-greeting-claude-code.txt"
+		plugin_name="Claude-aidevops plugin"
+		# shellcheck disable=SC2088
+		global_config_path="~/.config/Claude/Claude.json"
```

> ![medium](https://www.gstatic.com/codereviewagent/medium-priority.svg)
> 
> The `global_config_path` for `claude-code` is set to `~/.config/Claude/Claude.json`, but the `_generate_agents_claude` function (line 491) uses `~/.claude/settings.json`. This discrepancy should be resolved so the AI model is directed to the correct configuration file. Per repository rules, ensure consistency for similar patterns throughout the file.
> 
> ```suggestion
> 		global_config_path="~/.claude/settings.json"
> ```
> 
> <details>
> <summary>References</summary>
> 
> 1. When applying a fix for a specific pattern, also fix other instances of the same pattern in the file for consistency, even if they are not part of the original diff.
> </details>

#### gemini-code-assist on `.agents/scripts/generate-runtime-config.sh:159`

[View inline comment](https://github.com/marcusquinn/aidevops/pull/20612#discussion_r3132657207)

```diff
+		# shellcheck disable=SC2088
+		cache_path="~/.aidevops/cache/session-greeting-claude-code.txt"
+		plugin_name="Claude-aidevops plugin"
+		# shellcheck disable=SC2088
+		global_config_path="~/.config/Claude/Claude.json"
+		;;
+	*)
+		# shellcheck disable=SC2088
+		cache_path="~/.aidevops/cache/session-greeting-${runtime_id}.txt"
+		plugin_name="${runtime_id}-aidevops plugin"
+		# shellcheck disable=SC2088
+		global_config_path="~/.config/Claude/Claude.json"
```

> ![medium](https://www.gstatic.com/codereviewagent/medium-priority.svg)
> 
> The `global_config_path` in the fallback case (`*`) is hardcoded to the Claude configuration path. This appears to be a copy-paste error and should be parameterized to support other runtimes correctly.
> 
> ```suggestion
> 		global_config_path="~/.config/${runtime_id}/${runtime_id}.json"
> ```

### PR review summaries

#### gemini-code-assist[bot] review summary

[View review](https://github.com/marcusquinn/aidevops/pull/20612#pullrequestreview-4164571553)

> ## Code Review
> 
> This pull request refactors the generation of runtime-specific AGENTS.md files by introducing a shared helper function, _generate_greeting_agents_md, which parameterizes greeting templates and configuration paths for runtimes like OpenCode and Claude Code. It also updates the Claude Code configuration logic to ensure necessary directories exist and to populate a greeting cache. Feedback indicates a discrepancy in the global configuration path for Claude Code and a hardcoded path in the fallback case of the runtime selection logic that should be parameterized.
<!-- aidevops:sig -->
---
[aidevops.sh](https://aidevops.sh) v3.8.96 automated scan.

--- a/.agents/scripts/generate-runtime-config.sh
+++ b/.agents/scripts/generate-runtime-config.sh
@@ -156,7 +156,7 @@ _generate_greeting_agents_md() {
 		cache_path="~/.aidevops/cache/session-greeting-${runtime_id}.txt"
 		plugin_name="${runtime_id}-aidevops plugin"
 		# shellcheck disable=SC2088
-		global_config_path="~/.config/Claude/Claude.json"
+		global_config_path="~/.config/${runtime_id}/${runtime_id}.json"
 		;;
 	esac
 

